neqo-transport/src/connection/tests/migration.rs

```diff
@@ -62,27 +64,223 @@ const fn new_port(a: SocketAddr) -> SocketAddr {
     SocketAddr::new(a.ip(), port)
 }
 
+fn new_address_and_port(a: SocketAddr) -> SocketAddr {
+    let ip = match a.ip() {
+        IpAddr::V4(ip) => IpAddr::V4(Ipv4Addr::from(ip.octets().map(|b| b.overflowing_add(11).0))),
+        IpAddr::V6(ip) => IpAddr::V6(Ipv6Addr::from(ip.octets().map(|b| b.overflowing_add(11).0))),
+    };
+    SocketAddr::new(ip, new_port(a).port())
+}
+
 fn change_source_port(d: &Datagram) -> Datagram {
     Datagram::new(new_port(d.source()), d.destination(), d.tos(), &d[..])
 }
 
-#[test]
-fn rebinding_port() {
-    let mut client = default_client();
-    let mut server = default_server();
-    connect_force_idle(&mut client, &mut server);
+fn change_source_ip_and_port(d: &Datagram) -> Datagram {
+    Datagram::new(
+        new_address_and_port(d.source()),
+        d.destination(),
+        d.tos(),
+        &d[..],
+    )
+}
 
-    let dgram = send_something(&mut client, now());
-    let dgram = change_source_port(&dgram);
+fn assert_path_challenge(
+    c: &Connection,
+    d: &Datagram,
+    before: &FrameStats,
+    mod_saddr: fn(SocketAddr) -> SocketAddr,
+    padded: bool,
+) {
+    let after = c.stats().frame_tx;
+    assert_eq!(after.path_challenge, before.path_challenge + 1);
+    assert_eq!(d.source(), DEFAULT_ADDR);
+    assert_eq!(d.destination(), mod_saddr(DEFAULT_ADDR));
+    if padded {
+        assert!(d.len() >= MIN_INITIAL_PACKET_SIZE);
+    } else {
+        assert!(d.len() < MIN_INITIAL_PACKET_SIZE);
+    }
+}
 
-    server.process_input(&dgram, now());
-    // Have the server send something so that it generates a packet.
-    let stream_id = server.stream_create(StreamType::UniDi).unwrap();
-    server.stream_close_send(stream_id).unwrap();
+const fn id(a: SocketAddr) -> SocketAddr {
+    a
+}
```

Comment:
Where is this used? Could you use the `std::convert::identity` function instead?

https://doc.rust-lang.org/std/convert/fn.identity.html